packages/components/transfer/src/transfer.ts

```diff
@@ -1,68 +1,118 @@
-// @ts-nocheck
-import { CHANGE_EVENT } from '@element-plus/constants'
+import { isNil } from 'lodash-unified'
+import {
+  buildProps,
+  definePropType,
+  isArray,
+  mutable,
+} from '@element-plus/utils'
+import { CHANGE_EVENT, UPDATE_MODEL_EVENT } from '@element-plus/constants'
 
-import type { VNode } from 'vue'
+import type { ExtractPropTypes, h as H, VNode } from 'vue'
+import type Transfer from './transfer.vue'
 
-export { CHANGE_EVENT }
+export type TransferKey = string | number
+export type TransferDirection = 'left' | 'right'
 
-export type Key = string | number
+export interface TransferDataItem {
```

Comment:
This type is best marked directly as object

```ts
type TransferDataItem = object
```

If used externally the `render-content` and `filter-method` Api will restrict the type base on Record<string, any>.

does not work

```ts
interface Option {
  id?: number | string
  name?: string
}
```

Reply:
agree! I missed this point.

Reply:
@tolking How about `type TransferDataItem = Record<string, any>`? Directly specifying it directly as `object`, I need to additionally handle this ts type error internally:
`No index signature with a parameter of type 'string' was found on type '{}'`